### DISCUSSION TITLE
Logging to socket, the docs indicate how-to, but was it ever implemented?

### DISCUSSION THREAD

**Discussion by NikolaiVChr** [JSBSim-Team/jsbsim#986] [Q&A]
In other words, can I do this?
```xml
<output name="localhost" type="socket" port="7000" protocol="UDP" rate="500">
 <property> simulation/sim-time-sec </property>
 <property> attitude/pitch-rad </property>
</output>
```

Also, can I log to a socket and a file at the same time (different properties).

**Comment by NikolaiVChr** [comment]
PS. The docs where I found this is: https://jsbsim-team.github.io/jsbsim/classJSBSim_1_1FGOutput.html

**Comment by bcoconni** [comment]
Hi @NikolaiVChr,

To answer the question in the title: yes the UDP protocol is implemented.
 

> In other words, can I do this?
> 
> ```
> <output name="localhost" type="socket" port="7000" protocol="UDP" rate="500">
>  <property> simulation/sim-time-sec </property>
>  <property> attitude/pitch-rad </property>
> </output>
> ```

Yes but check that the rate is consistent with the simulation time step (120Hz by default). For a rate of 500, you should have a simulated time step of 0.002s.

> Also, can I log to a socket and a file at the same time (different properties).

Yes, JSBSim allows using as many output as you can fit in your computer memory and disk bandwidth.
